plugins/aws/resoto_plugin_aws/accountcollector.py

```diff
@@ -1244,6 +1249,62 @@ def collect_instances(self, region: AWSRegion, graph: Graph) -> None:
             except botocore.exceptions.ClientError:
                 log.exception(f"Some boto3 call failed on resource {instance} - skipping")
 
+    @metrics_collect_lambda_functions.time()  # type: ignore
+    def collect_lambda_functions(self, region: AWSRegion, graph: Graph) -> None:
+        log.info(f"Collecting AWS Lambda Functions in account {self.account.dname}, region {region.id}")
+        session = aws_session(self.account.id, self.account.role)
+        client = session.client("lambda", region_name=region.id)
+
+        for function in paginate(client.list_functions):
+            name = function["FunctionName"]
+            arn = function["FunctionArn"]
+
+            log.debug(f"Found AWS Lambda Function {name}")
+
+            tags_response = client.list_tags(Resource=arn)
+            tags = tags_response.get("Tags", [])
+
+            lambda_function = AWSLambdaFunction(
+                arn,
+                tags,
+                account=self.account,
+                region=region,
+                name=name,
+                arn=arn,
+            )
+
+            lambda_function.runtime = function["Runtime"]
+            lambda_function.code_size = int(function["CodeSize"])
+            lambda_function.memory_size = function["MemorySize"]
+            lambda_function.mtime = function["LastModified"]
+            lambda_function.role = function["Role"]
+            lambda_function.kms_key_arn = function.get("KmsKeyArn")
+
+            vpc_config = function.get("VpcConfig", {})
+
+            vpc_id = vpc_config.get("VpcId")
+            if vpc_id:
+                vpc = graph.search_first("id", vpc_id)
+                if vpc:
+                    graph.add_edge(vpc, lambda_function)
+                    graph.add_edge(vpc, lambda_function, edge_type=EdgeType.delete)
+
+            for subnet_id in vpc_config.get("SubnetIds", {}):
+                subnet = graph.search_first("id", subnet_id)
+                if subnet:
+                    graph.add_edge(subnet, lambda_function)
+                    graph.add_edge(subnet, lambda_function, edge_type=EdgeType.delete)
+
+            for security_group_id in vpc_config.get("SecurityGroupIds", {}):
+                security_group = graph.search_first("id", security_group_id)
+                if security_group:
+                    graph.add_edge(security_group, lambda_function)
+                    graph.add_edge(security_group, lambda_function, edge_type=EdgeType.delete)
+
+            # TODO get ENIs
```

Comment:
Is this a left over? Where would the ENIs come from?

Reply:
That was a note for me. Sadly it seems there's no clean way to get them. Although the IPs get assigned the only clear relation I could find in the API was the description of the ENI, which didn't seem like a scalable way to query it.